legal-api/src/legal_api/services/document_meta.py

```diff
@@ -34,136 +33,236 @@ class ReportType(Enum):
+        if self.is_bcomp() and self.is_completed():
+            reports.append(
+                self.create_report_object(
+                    'Notice of Articles',
+                    self.get_general_filename('Notice of Articles'),
+                    DocumentMetaService.ReportType.NOTICE_OF_ARTICLES.value
+                )
+            )
+
+        return reports
+
+    def get_cod_reports(self):
+        """Return change of director meta object(s)."""
+        reports = [
+            self.create_report_object(
+                'Director Change',
+                self.get_general_filename('Director Change')
+            )
+        ]
+
+        # when BCOMP filing is completed, also return NOA
+        if self.is_bcomp() and self.is_completed():
+            reports.append(
+                self.create_report_object(
+                    'Notice of Articles',
+                    self.get_general_filename('Notice of Articles'),
+                    DocumentMetaService.ReportType.NOTICE_OF_ARTICLES.value
+                )
+            )
+
+        return reports
+
+    def get_con_reports(self):
+        """Return change of name object(s)."""
+        reports = [
+            self.create_report_object(
+                'Legal Name Change',
+                self.get_general_filename('Legal Name Change')
+            )
+        ]
+
+        # when BCOMP filing is completed, also return NOA
+        if self.is_bcomp() and self.is_completed():
+            reports.append(
+                self.create_report_object(
+                    'Notice of Articles',
+                    self.get_general_filename('Notice of Articles'),
+                    DocumentMetaService.ReportType.NOTICE_OF_ARTICLES.value
+                )
+            )
+
+        return reports
+
+    def get_special_resolution_reports(self):
+        """Return special resolution meta object(s)."""
+        reports = []
+
+        if self.is_completed():
+            reports.append(
+                self.create_report_object(
+                    'Special Resolution',
+                    self.get_general_filename('Special Resolution')
+                )
+            )
+
+        return reports
+
+    def get_voluntary_dissolution_reports(self):
+        """Return voluntary dissolution meta object(s)."""
+        reports = []
+
+        if self.is_completed():
+            reports.append(
+                self.create_report_object(
+                    'Voluntary Dissolution',
+                    self.get_general_filename('Voluntary Dissolution')
+                )
+            )
+
+        return reports
+
+    def get_correction_reports(self):  # pylint: disable=no-self-use
+        """Return correction meta object(s)."""
+        # FUTURE: return applicable documents
+        return []
+
+    def get_alteration_reports(self):  # pylint: disable=no-self-use
+        """Return alteration meta object(s)."""
+        # FUTURE: return applicable documents
```

Comment:
I think this should return NOA

Reply:
Perhaps, but it's future and not in scope for this ticket.